src/main/resources/diagram_git_starship.puml

```diff
@@ -2,132 +2,181 @@
 
 package rape.brutal.gitstarship {
 
-    package ships {
+    class Universe {
+        -int kills;
+        -int moneySpends;
+        +void startUniverse();
+        +void winState();
+        +void loseState();
+        +void getStats();
+    }
 
-        Starship <|-- GitStarship
-        Starship <|-- PirateStarship
+    class Storage {
+        -HashMap<String, IStorageable> items;
+        +void putItem(IStorageable iStorageable);
+        +void getItem(String itemName);
+        -int capacity;
+    }
 
-        abstract class Starship {
-            #Engine engine;
-            #Hull hull;
-            #Storage<IStoreable> storage;
+    package trade {
 
-            +boolean putInStorage(IStoreable iStoreable);
-            +IStoreable getFromStorage(String iStoreableName);
+        abstract class Trader {
+            -int cash;
+            -Storage storage;
+        }
 
-            -void flyTo(IReachable iReachable);
+        class AverageTrader extends Trader {
+            +void sell(int cash, String itemName);
+            +void buy(int cash, String itemName);
+        }
 
-            +void trade(Trader trader);
+        class Mechanic extends Trader {
+            +void repair(int cash, String itemName);
+            +void upgrade(int cash, String itemName);
+        }
 
-            +void setDamage(int damage);
-            +int getHitPoints();
+    }
 
-            +void attack(IAttackable iAttackable);
-        }
+    package events {
 
-        class GitStarship {
-            -Railgun railgun;
-            -Turret rightTurret;
-            -Turret leftTurret;
+        +abstract class Event {
+            #Universe universe;
+            #Location location;
+            #GitStarship gitStarship;
+
+            +void start(GitStarship gitStarship);
         }
 
-        class PirateStarship {
-            -Turret rightTurret;
+        +class Battle {
+            -Universe universe;
+            -GitStarship gitStarship;
+            -ArrayList<Starship> atackers;
+
+            +void startBattle();
         }
 
     }
 
-    package stuff {
-
-        ValuableItem <|-- TurretMagazineCase
-        ValuableItem <|-- MetalShell
+    interface ILandable {
+        +void land(GitStarship gitStarship);
+    }
 
-        class TurretMagazineCase {
+    package locations {
 
+        +abstract class Location {
+            #GitStarship gitStarship;
         }
 
-        class MetalShell {
+        +class Planet extends Location implements ILandable {
+            -Trader trader;
+            -Storage storage;
+        }
 
+        +class Asteroid extends Location  implements ILandable {
+            -Trader trader;
+            -Storage storage;
         }
 
-        abstract class ValuableItem {
-            -int value;
+        +class OpenSpace extends Location {
+            -ArrayList<Starship> ships;
```

Comment:
Why there are no possibility to find storage in open space?

It can be a pieces of blown Starship or life-saving capsule etc.